cannot load such file -- mysql2/2.4/mysql2 (LoadError)`

I am new on Ruby on Rails. My Ruby version is ruby2.4.1p111 (2017-03-22 revision 58053) [x64-mingw32].

I have created a new project using rails new sample_cms -d mysql command.

The command rails server is throwing following error.

C:/Ruby24-x64/lib/ruby/gems/2.4.0/gems/mysql2-0.4.6-x64-mingw32/lib/mysql2/mysql2.rb:2:inrequire': cannot load such file -- mysql2/2.4/mysql2 (LoadError)`
I have installed mysql and mysql Connector C. I have installed the mysql2 gem like that.

gem install mysql2 —platform=ruby — ‘—with-mysql-dir="C:\Program Files\MySQL\MySQL Connector C 6.1"’
I have uninstall and install the mysql2 gem but still facing the issue. Here is the directory which may have issue.

![27505154-7b8f3f4e-58b2-11e7-8c8a-66980553443c](https://user-images.githubusercontent.com/29669465/27507947-932227b0-58f4-11e7-9b5d-11d8707d0d4b.png)

Here is the gem file entry for mysql2

# Bundle edge Rails instead: gem 'rails', github: 'rails/rails'
gem 'rails', '~> 5.1.1'
# Use mysql as the database for Active Record
gem 'mysql2', '>= 0.3.18', '< 0.5'
# Use Puma as the app server
gem 'puma', '~> 3.7'
Any Help?

I'm also having problem in using mysql2 gem in Ruby 2.4. The binary is not listed inside the gem directory, and googling takes me here. I have tried downloading the source code, building the gem by using `**gem build** _mysql2.gemspec_`, then install it by providing the mysql connector location -since that I'm using 32-bit version of Ruby inside 64-bit Windows-, and then I got this error. Any idea on how to compile it properly?

```
E:\mysql2-0.4.6>gem install .\mysql2-0.4.6.gem --platform ruby -- '--with-mysql-dir="C:/Program Files (x86)/MySQL/MySQL Connector C 6.1"'
Temporarily enhancing PATH for MSYS/MINGW...
Building native extensions with: '--with-mysql-dir="C:/Program Files (x86)/MySQL/MySQL Connector C 6.1"'
This could take a while...
ERROR:  Error installing .\mysql2-0.4.6.gem:
        ERROR: Failed to build gem native extension.

    current directory: E:/ruby24x86/lib/ruby/gems/2.4.0/gems/mysql2-0.4.6/ext/mysql2
E:/ruby24x86/bin/ruby.exe -r ./siteconf20170625-6748-g1m887.rb extconf.rb --with-mysql-dir="C:/Program Files (x86)/MySQL/MySQL Connector C 6.1"
checking for rb_absint_size()... yes
checking for rb_absint_singlebit_p()... yes
checking for ruby/thread.h... yes
checking for rb_thread_call_without_gvl() in ruby/thread.h... yes
checking for rb_thread_blocking_region()... no
checking for rb_wait_for_single_fd()... yes
checking for rb_hash_dup()... yes
checking for rb_intern3()... yes
checking for rb_big_cmp()... yes
-----
Using --with-mysql-dir=C:/Program Files (x86)/MySQL/MySQL Connector C 6.1
-----
checking for mysql.h... yes
checking for SSL_MODE_DISABLED in mysql.h... yes
checking for SSL_MODE_PREFERRED in mysql.h... yes
checking for SSL_MODE_REQUIRED in mysql.h... yes
checking for SSL_MODE_VERIFY_CA in mysql.h... yes
checking for SSL_MODE_VERIFY_IDENTITY in mysql.h... yes
checking for errmsg.h... yes
checking for mysqld_error.h... yes
*** extconf.rb failed ***
Could not create Makefile due to some reason, probably lack of necessary libraries and/or headers.  Check the mkmf.log file for more details.  You may need configuration options.

Provided configuration options:
        --with-opt-dir
        --without-opt-dir
        --with-opt-include
        --without-opt-include=${opt-dir}/include
        --with-opt-lib
        --without-opt-lib=${opt-dir}/lib
        --with-make-prog
        --without-make-prog
        --srcdir=.
        --curdir
        --ruby=E:/ruby24x86/bin/$(RUBY_BASE_NAME)
        --with-mysql-dir
        --with-mysql-include
        --without-mysql-include=${mysql-dir}/include
        --with-mysql-lib
        --without-mysql-lib=${mysql-dir}/lib
        --with-sanitize
        --without-sanitize
E:/ruby24x86/lib/ruby/gems/2.4.0/gems/rake-12.0.0/lib/rake/task_manager.rb:58:in `[]': Don't know how to build task 'E:/ruby24x86/lib/ruby/gems/2.4.0/gems/mysql2-0.4.6/support/libmysql.def' (see --tasks) (RuntimeError)
        from E:/ruby24x86/lib/ruby/gems/2.4.0/gems/rake-12.0.0/lib/rake/task.rb:61:in `lookup_prerequisite'
        from E:/ruby24x86/lib/ruby/gems/2.4.0/gems/rake-12.0.0/lib/rake/task.rb:57:in `block in prerequisite_tasks'
        from E:/ruby24x86/lib/ruby/gems/2.4.0/gems/rake-12.0.0/lib/rake/task.rb:57:in `map'
        from E:/ruby24x86/lib/ruby/gems/2.4.0/gems/rake-12.0.0/lib/rake/task.rb:57:in `prerequisite_tasks'
        from E:/ruby24x86/lib/ruby/gems/2.4.0/gems/rake-12.0.0/lib/rake/task.rb:214:in `invoke_prerequisites'
        from E:/ruby24x86/lib/ruby/gems/2.4.0/gems/rake-12.0.0/lib/rake/task.rb:193:in `block in invoke_with_call_chain'

        from E:/ruby24x86/lib/ruby/2.4.0/monitor.rb:214:in `mon_synchronize'
        from E:/ruby24x86/lib/ruby/gems/2.4.0/gems/rake-12.0.0/lib/rake/task.rb:187:in `invoke_with_call_chain'
        from E:/ruby24x86/lib/ruby/gems/2.4.0/gems/rake-12.0.0/lib/rake/task.rb:180:in `invoke'
        from extconf.rb:197:in `<main>'

To see why this extension failed to compile, please check the mkmf.log which can be found here:

  E:/ruby24x86/lib/ruby/gems/2.4.0/extensions/x86-mingw32/2.4.0/mysql2-0.4.6/mkmf.log

extconf failed, exit code 1

Gem files will remain installed in E:/ruby24x86/lib/ruby/gems/2.4.0/gems/mysql2-0.4.6 for inspection.
Results logged to E:/ruby24x86/lib/ruby/gems/2.4.0/extensions/x86-mingw32/2.4.0/mysql2-0.4.6/gem_make.out
```

The content of mkmf.log file is shown as below:

```
have_func: checking for rb_absint_size()... -------------------- yes

"i686-w64-mingw32-gcc -o conftest.exe -IE:/ruby24x86/include/ruby-2.4.0/i386-mingw32 -IE:/ruby24x86/include/ruby-2.4.0/ruby/backward -IE:/ruby24x86/include/ruby-2.4.0 -I. -D_FORTIFY_SOURCE=2 -D__USE_MINGW_ANSI_STDIO=1 -DFD_SETSIZE=2048 -IC:/msys64/mingw32/lib/libffi-3.2.1/include -D_WIN32_WINNT=0x0501 -D__MINGW_USE_VC2005_COMPAT -D_FILE_OFFSET_BITS=64  -march=i686 -mtune=generic -O2 -pipe -IC:/msys64/mingw32/lib/libffi-3.2.1/include conftest.c  -L. -LE:/ruby24x86/lib -L. -pipe     -lmsvcrt-ruby240  -lgmp -lshell32 -lws2_32 -liphlpapi -limagehlp -lshlwapi  "
checked program was:
/* begin */
1: #include "ruby.h"
2: 
3: #include <winsock2.h>
4: #include <windows.h>
5: int main(int argc, char **argv)
6: {
7:   return 0;
8: }
/* end */

"i686-w64-mingw32-gcc -o conftest.exe -IE:/ruby24x86/include/ruby-2.4.0/i386-mingw32 -IE:/ruby24x86/include/ruby-2.4.0/ruby/backward -IE:/ruby24x86/include/ruby-2.4.0 -I. -D_FORTIFY_SOURCE=2 -D__USE_MINGW_ANSI_STDIO=1 -DFD_SETSIZE=2048 -IC:/msys64/mingw32/lib/libffi-3.2.1/include -D_WIN32_WINNT=0x0501 -D__MINGW_USE_VC2005_COMPAT -D_FILE_OFFSET_BITS=64  -march=i686 -mtune=generic -O2 -pipe -IC:/msys64/mingw32/lib/libffi-3.2.1/include conftest.c  -L. -LE:/ruby24x86/lib -L. -pipe     -lmsvcrt-ruby240  -lgmp -lshell32 -lws2_32 -liphlpapi -limagehlp -lshlwapi  "
checked program was:
/* begin */
 1: #include "ruby.h"
 2: 
 3: #include <winsock2.h>
 4: #include <windows.h>
 5: 
 6: /*top*/
 7: extern int t(void);
 8: int main(int argc, char **argv)
 9: {
10:   if (argc > 1000000) {
11:     printf("%p", &t);
12:   }
13: 
14:   return 0;
15: }
16: int t(void) { void ((*volatile p)()); p = (void ((*)()))rb_absint_size; return !p; }
/* end */

--------------------

have_func: checking for rb_absint_singlebit_p()... -------------------- yes

"i686-w64-mingw32-gcc -o conftest.exe -IE:/ruby24x86/include/ruby-2.4.0/i386-mingw32 -IE:/ruby24x86/include/ruby-2.4.0/ruby/backward -IE:/ruby24x86/include/ruby-2.4.0 -I. -D_FORTIFY_SOURCE=2 -D__USE_MINGW_ANSI_STDIO=1 -DFD_SETSIZE=2048 -IC:/msys64/mingw32/lib/libffi-3.2.1/include -D_WIN32_WINNT=0x0501 -D__MINGW_USE_VC2005_COMPAT -D_FILE_OFFSET_BITS=64  -march=i686 -mtune=generic -O2 -pipe -IC:/msys64/mingw32/lib/libffi-3.2.1/include conftest.c  -L. -LE:/ruby24x86/lib -L. -pipe     -lmsvcrt-ruby240  -lgmp -lshell32 -lws2_32 -liphlpapi -limagehlp -lshlwapi  "
checked program was:
/* begin */
 1: #include "ruby.h"
 2: 
 3: #include <winsock2.h>
 4: #include <windows.h>
 5: 
 6: /*top*/
 7: extern int t(void);
 8: int main(int argc, char **argv)
 9: {
10:   if (argc > 1000000) {
11:     printf("%p", &t);
12:   }
13: 
14:   return 0;
15: }
16: int t(void) { void ((*volatile p)()); p = (void ((*)()))rb_absint_singlebit_p; return !p; }
/* end */

--------------------

have_header: checking for ruby/thread.h... -------------------- yes

"i686-w64-mingw32-gcc -E -IE:/ruby24x86/include/ruby-2.4.0/i386-mingw32 -IE:/ruby24x86/include/ruby-2.4.0/ruby/backward -IE:/ruby24x86/include/ruby-2.4.0 -I. -D_FORTIFY_SOURCE=2 -D__USE_MINGW_ANSI_STDIO=1 -DFD_SETSIZE=2048 -IC:/msys64/mingw32/lib/libffi-3.2.1/include -D_WIN32_WINNT=0x0501 -D__MINGW_USE_VC2005_COMPAT -D_FILE_OFFSET_BITS=64  -march=i686 -mtune=generic -O2 -pipe -IC:/msys64/mingw32/lib/libffi-3.2.1/include  conftest.c -o conftest.i"
checked program was:
/* begin */
1: #include "ruby.h"
2: 
3: #include <winsock2.h>
4: #include <windows.h>
5: #include <ruby/thread.h>
/* end */

--------------------

have_func: checking for rb_thread_call_without_gvl() in ruby/thread.h... -------------------- yes

"i686-w64-mingw32-gcc -o conftest.exe -IE:/ruby24x86/include/ruby-2.4.0/i386-mingw32 -IE:/ruby24x86/include/ruby-2.4.0/ruby/backward -IE:/ruby24x86/include/ruby-2.4.0 -I. -D_FORTIFY_SOURCE=2 -D__USE_MINGW_ANSI_STDIO=1 -DFD_SETSIZE=2048 -IC:/msys64/mingw32/lib/libffi-3.2.1/include -D_WIN32_WINNT=0x0501 -D__MINGW_USE_VC2005_COMPAT -D_FILE_OFFSET_BITS=64  -march=i686 -mtune=generic -O2 -pipe -IC:/msys64/mingw32/lib/libffi-3.2.1/include conftest.c  -L. -LE:/ruby24x86/lib -L. -pipe     -lmsvcrt-ruby240  -lgmp -lshell32 -lws2_32 -liphlpapi -limagehlp -lshlwapi  "
checked program was:
/* begin */
 1: #include "ruby.h"
 2: 
 3: #include <winsock2.h>
 4: #include <windows.h>
 5: #include <ruby/thread.h>
 6: 
 7: /*top*/
 8: extern int t(void);
 9: int main(int argc, char **argv)
10: {
11:   if (argc > 1000000) {
12:     printf("%p", &t);
13:   }
14: 
15:   return 0;
16: }
17: int t(void) { void ((*volatile p)()); p = (void ((*)()))rb_thread_call_without_gvl; return !p; }
/* end */

--------------------

have_func: checking for rb_thread_blocking_region()... -------------------- no

"i686-w64-mingw32-gcc -o conftest.exe -IE:/ruby24x86/include/ruby-2.4.0/i386-mingw32 -IE:/ruby24x86/include/ruby-2.4.0/ruby/backward -IE:/ruby24x86/include/ruby-2.4.0 -I. -D_FORTIFY_SOURCE=2 -D__USE_MINGW_ANSI_STDIO=1 -DFD_SETSIZE=2048 -IC:/msys64/mingw32/lib/libffi-3.2.1/include -D_WIN32_WINNT=0x0501 -D__MINGW_USE_VC2005_COMPAT -D_FILE_OFFSET_BITS=64  -march=i686 -mtune=generic -O2 -pipe -IC:/msys64/mingw32/lib/libffi-3.2.1/include conftest.c  -L. -LE:/ruby24x86/lib -L. -pipe     -lmsvcrt-ruby240  -lgmp -lshell32 -lws2_32 -liphlpapi -limagehlp -lshlwapi  "
conftest.c: In function 't':
conftest.c:16:57: error: 'rb_thread_blocking_region' undeclared (first use in this function)
 int t(void) { void ((*volatile p)()); p = (void ((*)()))rb_thread_blocking_region; return !p; }
                                                         ^~~~~~~~~~~~~~~~~~~~~~~~~
conftest.c:16:57: note: each undeclared identifier is reported only once for each function it appears in
checked program was:
/* begin */
 1: #include "ruby.h"
 2: 
 3: #include <winsock2.h>
 4: #include <windows.h>
 5: 
 6: /*top*/
 7: extern int t(void);
 8: int main(int argc, char **argv)
 9: {
10:   if (argc > 1000000) {
11:     printf("%p", &t);
12:   }
13: 
14:   return 0;
15: }
16: int t(void) { void ((*volatile p)()); p = (void ((*)()))rb_thread_blocking_region; return !p; }
/* end */

"i686-w64-mingw32-gcc -o conftest.exe -IE:/ruby24x86/include/ruby-2.4.0/i386-mingw32 -IE:/ruby24x86/include/ruby-2.4.0/ruby/backward -IE:/ruby24x86/include/ruby-2.4.0 -I. -D_FORTIFY_SOURCE=2 -D__USE_MINGW_ANSI_STDIO=1 -DFD_SETSIZE=2048 -IC:/msys64/mingw32/lib/libffi-3.2.1/include -D_WIN32_WINNT=0x0501 -D__MINGW_USE_VC2005_COMPAT -D_FILE_OFFSET_BITS=64  -march=i686 -mtune=generic -O2 -pipe -IC:/msys64/mingw32/lib/libffi-3.2.1/include conftest.c  -L. -LE:/ruby24x86/lib -L. -pipe     -lmsvcrt-ruby240  -lgmp -lshell32 -lws2_32 -liphlpapi -limagehlp -lshlwapi  "
C:\Users\jianh\AppData\Local\Temp\cc8qbCYP.o:conftest.c:(.text+0x4): undefined reference to `rb_thread_blocking_region'
collect2.exe: error: ld returned 1 exit status
checked program was:
/* begin */
 1: #include "ruby.h"
 2: 
 3: #include <winsock2.h>
 4: #include <windows.h>
 5: 
 6: /*top*/
 7: extern int t(void);
 8: int main(int argc, char **argv)
 9: {
10:   if (argc > 1000000) {
11:     printf("%p", &t);
12:   }
13: 
14:   return 0;
15: }
16: extern void rb_thread_blocking_region();
17: int t(void) { rb_thread_blocking_region(); return 0; }
/* end */

--------------------

have_func: checking for rb_wait_for_single_fd()... -------------------- yes

"i686-w64-mingw32-gcc -o conftest.exe -IE:/ruby24x86/include/ruby-2.4.0/i386-mingw32 -IE:/ruby24x86/include/ruby-2.4.0/ruby/backward -IE:/ruby24x86/include/ruby-2.4.0 -I. -D_FORTIFY_SOURCE=2 -D__USE_MINGW_ANSI_STDIO=1 -DFD_SETSIZE=2048 -IC:/msys64/mingw32/lib/libffi-3.2.1/include -D_WIN32_WINNT=0x0501 -D__MINGW_USE_VC2005_COMPAT -D_FILE_OFFSET_BITS=64  -march=i686 -mtune=generic -O2 -pipe -IC:/msys64/mingw32/lib/libffi-3.2.1/include conftest.c  -L. -LE:/ruby24x86/lib -L. -pipe     -lmsvcrt-ruby240  -lgmp -lshell32 -lws2_32 -liphlpapi -limagehlp -lshlwapi  "
conftest.c: In function 't':
conftest.c:16:57: error: 'rb_wait_for_single_fd' undeclared (first use in this function)
 int t(void) { void ((*volatile p)()); p = (void ((*)()))rb_wait_for_single_fd; return !p; }
                                                         ^~~~~~~~~~~~~~~~~~~~~
conftest.c:16:57: note: each undeclared identifier is reported only once for each function it appears in
checked program was:
/* begin */
 1: #include "ruby.h"
 2: 
 3: #include <winsock2.h>
 4: #include <windows.h>
 5: 
 6: /*top*/
 7: extern int t(void);
 8: int main(int argc, char **argv)
 9: {
10:   if (argc > 1000000) {
11:     printf("%p", &t);
12:   }
13: 
14:   return 0;
15: }
16: int t(void) { void ((*volatile p)()); p = (void ((*)()))rb_wait_for_single_fd; return !p; }
/* end */

"i686-w64-mingw32-gcc -o conftest.exe -IE:/ruby24x86/include/ruby-2.4.0/i386-mingw32 -IE:/ruby24x86/include/ruby-2.4.0/ruby/backward -IE:/ruby24x86/include/ruby-2.4.0 -I. -D_FORTIFY_SOURCE=2 -D__USE_MINGW_ANSI_STDIO=1 -DFD_SETSIZE=2048 -IC:/msys64/mingw32/lib/libffi-3.2.1/include -D_WIN32_WINNT=0x0501 -D__MINGW_USE_VC2005_COMPAT -D_FILE_OFFSET_BITS=64  -march=i686 -mtune=generic -O2 -pipe -IC:/msys64/mingw32/lib/libffi-3.2.1/include conftest.c  -L. -LE:/ruby24x86/lib -L. -pipe     -lmsvcrt-ruby240  -lgmp -lshell32 -lws2_32 -liphlpapi -limagehlp -lshlwapi  "
checked program was:
/* begin */
 1: #include "ruby.h"
 2: 
 3: #include <winsock2.h>
 4: #include <windows.h>
 5: 
 6: /*top*/
 7: extern int t(void);
 8: int main(int argc, char **argv)
 9: {
10:   if (argc > 1000000) {
11:     printf("%p", &t);
12:   }
13: 
14:   return 0;
15: }
16: extern void rb_wait_for_single_fd();
17: int t(void) { rb_wait_for_single_fd(); return 0; }
/* end */

--------------------

have_func: checking for rb_hash_dup()... -------------------- yes

"i686-w64-mingw32-gcc -o conftest.exe -IE:/ruby24x86/include/ruby-2.4.0/i386-mingw32 -IE:/ruby24x86/include/ruby-2.4.0/ruby/backward -IE:/ruby24x86/include/ruby-2.4.0 -I. -D_FORTIFY_SOURCE=2 -D__USE_MINGW_ANSI_STDIO=1 -DFD_SETSIZE=2048 -IC:/msys64/mingw32/lib/libffi-3.2.1/include -D_WIN32_WINNT=0x0501 -D__MINGW_USE_VC2005_COMPAT -D_FILE_OFFSET_BITS=64  -march=i686 -mtune=generic -O2 -pipe -IC:/msys64/mingw32/lib/libffi-3.2.1/include conftest.c  -L. -LE:/ruby24x86/lib -L. -pipe     -lmsvcrt-ruby240  -lgmp -lshell32 -lws2_32 -liphlpapi -limagehlp -lshlwapi  "
checked program was:
/* begin */
 1: #include "ruby.h"
 2: 
 3: #include <winsock2.h>
 4: #include <windows.h>
 5: 
 6: /*top*/
 7: extern int t(void);
 8: int main(int argc, char **argv)
 9: {
10:   if (argc > 1000000) {
11:     printf("%p", &t);
12:   }
13: 
14:   return 0;
15: }
16: int t(void) { void ((*volatile p)()); p = (void ((*)()))rb_hash_dup; return !p; }
/* end */

--------------------

have_func: checking for rb_intern3()... -------------------- yes

"i686-w64-mingw32-gcc -o conftest.exe -IE:/ruby24x86/include/ruby-2.4.0/i386-mingw32 -IE:/ruby24x86/include/ruby-2.4.0/ruby/backward -IE:/ruby24x86/include/ruby-2.4.0 -I. -D_FORTIFY_SOURCE=2 -D__USE_MINGW_ANSI_STDIO=1 -DFD_SETSIZE=2048 -IC:/msys64/mingw32/lib/libffi-3.2.1/include -D_WIN32_WINNT=0x0501 -D__MINGW_USE_VC2005_COMPAT -D_FILE_OFFSET_BITS=64  -march=i686 -mtune=generic -O2 -pipe -IC:/msys64/mingw32/lib/libffi-3.2.1/include conftest.c  -L. -LE:/ruby24x86/lib -L. -pipe     -lmsvcrt-ruby240  -lgmp -lshell32 -lws2_32 -liphlpapi -limagehlp -lshlwapi  "
conftest.c: In function 't':
conftest.c:16:57: error: 'rb_intern3' undeclared (first use in this function)
 int t(void) { void ((*volatile p)()); p = (void ((*)()))rb_intern3; return !p; }
                                                         ^~~~~~~~~~
conftest.c:16:57: note: each undeclared identifier is reported only once for each function it appears in
checked program was:
/* begin */
 1: #include "ruby.h"
 2: 
 3: #include <winsock2.h>
 4: #include <windows.h>
 5: 
 6: /*top*/
 7: extern int t(void);
 8: int main(int argc, char **argv)
 9: {
10:   if (argc > 1000000) {
11:     printf("%p", &t);
12:   }
13: 
14:   return 0;
15: }
16: int t(void) { void ((*volatile p)()); p = (void ((*)()))rb_intern3; return !p; }
/* end */

"i686-w64-mingw32-gcc -o conftest.exe -IE:/ruby24x86/include/ruby-2.4.0/i386-mingw32 -IE:/ruby24x86/include/ruby-2.4.0/ruby/backward -IE:/ruby24x86/include/ruby-2.4.0 -I. -D_FORTIFY_SOURCE=2 -D__USE_MINGW_ANSI_STDIO=1 -DFD_SETSIZE=2048 -IC:/msys64/mingw32/lib/libffi-3.2.1/include -D_WIN32_WINNT=0x0501 -D__MINGW_USE_VC2005_COMPAT -D_FILE_OFFSET_BITS=64  -march=i686 -mtune=generic -O2 -pipe -IC:/msys64/mingw32/lib/libffi-3.2.1/include conftest.c  -L. -LE:/ruby24x86/lib -L. -pipe     -lmsvcrt-ruby240  -lgmp -lshell32 -lws2_32 -liphlpapi -limagehlp -lshlwapi  "
checked program was:
/* begin */
 1: #include "ruby.h"
 2: 
 3: #include <winsock2.h>
 4: #include <windows.h>
 5: 
 6: /*top*/
 7: extern int t(void);
 8: int main(int argc, char **argv)
 9: {
10:   if (argc > 1000000) {
11:     printf("%p", &t);
12:   }
13: 
14:   return 0;
15: }
16: extern void rb_intern3();
17: int t(void) { rb_intern3(); return 0; }
/* end */

--------------------

have_func: checking for rb_big_cmp()... -------------------- yes

"i686-w64-mingw32-gcc -o conftest.exe -IE:/ruby24x86/include/ruby-2.4.0/i386-mingw32 -IE:/ruby24x86/include/ruby-2.4.0/ruby/backward -IE:/ruby24x86/include/ruby-2.4.0 -I. -D_FORTIFY_SOURCE=2 -D__USE_MINGW_ANSI_STDIO=1 -DFD_SETSIZE=2048 -IC:/msys64/mingw32/lib/libffi-3.2.1/include -D_WIN32_WINNT=0x0501 -D__MINGW_USE_VC2005_COMPAT -D_FILE_OFFSET_BITS=64  -march=i686 -mtune=generic -O2 -pipe -IC:/msys64/mingw32/lib/libffi-3.2.1/include conftest.c  -L. -LE:/ruby24x86/lib -L. -pipe     -lmsvcrt-ruby240  -lgmp -lshell32 -lws2_32 -liphlpapi -limagehlp -lshlwapi  "
checked program was:
/* begin */
 1: #include "ruby.h"
 2: 
 3: #include <winsock2.h>
 4: #include <windows.h>
 5: 
 6: /*top*/
 7: extern int t(void);
 8: int main(int argc, char **argv)
 9: {
10:   if (argc > 1000000) {
11:     printf("%p", &t);
12:   }
13: 
14:   return 0;
15: }
16: int t(void) { void ((*volatile p)()); p = (void ((*)()))rb_big_cmp; return !p; }
/* end */

--------------------

have_header: checking for mysql.h... -------------------- yes

"i686-w64-mingw32-gcc -E -IE:/ruby24x86/include/ruby-2.4.0/i386-mingw32 -IE:/ruby24x86/include/ruby-2.4.0/ruby/backward -IE:/ruby24x86/include/ruby-2.4.0 -I. "-IC:/Program Files (x86)/MySQL/MySQL Connector C 6.1/include" -D_FORTIFY_SOURCE=2 -D__USE_MINGW_ANSI_STDIO=1 -DFD_SETSIZE=2048 -IC:/msys64/mingw32/lib/libffi-3.2.1/include -D_WIN32_WINNT=0x0501 -D__MINGW_USE_VC2005_COMPAT -D_FILE_OFFSET_BITS=64  -march=i686 -mtune=generic -O2 -pipe -IC:/msys64/mingw32/lib/libffi-3.2.1/include  conftest.c -o conftest.i"
checked program was:
/* begin */
1: #include "ruby.h"
2: 
3: #include <winsock2.h>
4: #include <windows.h>
5: #include <mysql.h>
/* end */

--------------------

have_const: checking for SSL_MODE_DISABLED in mysql.h... -------------------- yes

"i686-w64-mingw32-gcc -IE:/ruby24x86/include/ruby-2.4.0/i386-mingw32 -IE:/ruby24x86/include/ruby-2.4.0/ruby/backward -IE:/ruby24x86/include/ruby-2.4.0 -I. "-IC:/Program Files (x86)/MySQL/MySQL Connector C 6.1/include" -D_FORTIFY_SOURCE=2 -D__USE_MINGW_ANSI_STDIO=1 -DFD_SETSIZE=2048 -IC:/msys64/mingw32/lib/libffi-3.2.1/include -D_WIN32_WINNT=0x0501 -D__MINGW_USE_VC2005_COMPAT -D_FILE_OFFSET_BITS=64  -march=i686 -mtune=generic -O2 -pipe -IC:/msys64/mingw32/lib/libffi-3.2.1/include   -c conftest.c"
checked program was:
/* begin */
1: #include "ruby.h"
2: 
3: #include <winsock2.h>
4: #include <windows.h>
5: #include <mysql.h>
6: 
7: /*top*/
8: typedef int conftest_type;
9: conftest_type conftestval = (int)SSL_MODE_DISABLED;
/* end */

--------------------

have_const: checking for SSL_MODE_PREFERRED in mysql.h... -------------------- yes

"i686-w64-mingw32-gcc -IE:/ruby24x86/include/ruby-2.4.0/i386-mingw32 -IE:/ruby24x86/include/ruby-2.4.0/ruby/backward -IE:/ruby24x86/include/ruby-2.4.0 -I. "-IC:/Program Files (x86)/MySQL/MySQL Connector C 6.1/include" -D_FORTIFY_SOURCE=2 -D__USE_MINGW_ANSI_STDIO=1 -DFD_SETSIZE=2048 -IC:/msys64/mingw32/lib/libffi-3.2.1/include -D_WIN32_WINNT=0x0501 -D__MINGW_USE_VC2005_COMPAT -D_FILE_OFFSET_BITS=64  -march=i686 -mtune=generic -O2 -pipe -IC:/msys64/mingw32/lib/libffi-3.2.1/include   -c conftest.c"
checked program was:
/* begin */
1: #include "ruby.h"
2: 
3: #include <winsock2.h>
4: #include <windows.h>
5: #include <mysql.h>
6: 
7: /*top*/
8: typedef int conftest_type;
9: conftest_type conftestval = (int)SSL_MODE_PREFERRED;
/* end */

--------------------

have_const: checking for SSL_MODE_REQUIRED in mysql.h... -------------------- yes

"i686-w64-mingw32-gcc -IE:/ruby24x86/include/ruby-2.4.0/i386-mingw32 -IE:/ruby24x86/include/ruby-2.4.0/ruby/backward -IE:/ruby24x86/include/ruby-2.4.0 -I. "-IC:/Program Files (x86)/MySQL/MySQL Connector C 6.1/include" -D_FORTIFY_SOURCE=2 -D__USE_MINGW_ANSI_STDIO=1 -DFD_SETSIZE=2048 -IC:/msys64/mingw32/lib/libffi-3.2.1/include -D_WIN32_WINNT=0x0501 -D__MINGW_USE_VC2005_COMPAT -D_FILE_OFFSET_BITS=64  -march=i686 -mtune=generic -O2 -pipe -IC:/msys64/mingw32/lib/libffi-3.2.1/include   -c conftest.c"
checked program was:
/* begin */
1: #include "ruby.h"
2: 
3: #include <winsock2.h>
4: #include <windows.h>
5: #include <mysql.h>
6: 
7: /*top*/
8: typedef int conftest_type;
9: conftest_type conftestval = (int)SSL_MODE_REQUIRED;
/* end */

--------------------

have_const: checking for SSL_MODE_VERIFY_CA in mysql.h... -------------------- yes

"i686-w64-mingw32-gcc -IE:/ruby24x86/include/ruby-2.4.0/i386-mingw32 -IE:/ruby24x86/include/ruby-2.4.0/ruby/backward -IE:/ruby24x86/include/ruby-2.4.0 -I. "-IC:/Program Files (x86)/MySQL/MySQL Connector C 6.1/include" -D_FORTIFY_SOURCE=2 -D__USE_MINGW_ANSI_STDIO=1 -DFD_SETSIZE=2048 -IC:/msys64/mingw32/lib/libffi-3.2.1/include -D_WIN32_WINNT=0x0501 -D__MINGW_USE_VC2005_COMPAT -D_FILE_OFFSET_BITS=64  -march=i686 -mtune=generic -O2 -pipe -IC:/msys64/mingw32/lib/libffi-3.2.1/include   -c conftest.c"
checked program was:
/* begin */
1: #include "ruby.h"
2: 
3: #include <winsock2.h>
4: #include <windows.h>
5: #include <mysql.h>
6: 
7: /*top*/
8: typedef int conftest_type;
9: conftest_type conftestval = (int)SSL_MODE_VERIFY_CA;
/* end */

--------------------

have_const: checking for SSL_MODE_VERIFY_IDENTITY in mysql.h... -------------------- yes

"i686-w64-mingw32-gcc -IE:/ruby24x86/include/ruby-2.4.0/i386-mingw32 -IE:/ruby24x86/include/ruby-2.4.0/ruby/backward -IE:/ruby24x86/include/ruby-2.4.0 -I. "-IC:/Program Files (x86)/MySQL/MySQL Connector C 6.1/include" -D_FORTIFY_SOURCE=2 -D__USE_MINGW_ANSI_STDIO=1 -DFD_SETSIZE=2048 -IC:/msys64/mingw32/lib/libffi-3.2.1/include -D_WIN32_WINNT=0x0501 -D__MINGW_USE_VC2005_COMPAT -D_FILE_OFFSET_BITS=64  -march=i686 -mtune=generic -O2 -pipe -IC:/msys64/mingw32/lib/libffi-3.2.1/include   -c conftest.c"
checked program was:
/* begin */
1: #include "ruby.h"
2: 
3: #include <winsock2.h>
4: #include <windows.h>
5: #include <mysql.h>
6: 
7: /*top*/
8: typedef int conftest_type;
9: conftest_type conftestval = (int)SSL_MODE_VERIFY_IDENTITY;
/* end */

--------------------

have_header: checking for errmsg.h... -------------------- yes

"i686-w64-mingw32-gcc -E -IE:/ruby24x86/include/ruby-2.4.0/i386-mingw32 -IE:/ruby24x86/include/ruby-2.4.0/ruby/backward -IE:/ruby24x86/include/ruby-2.4.0 -I. "-IC:/Program Files (x86)/MySQL/MySQL Connector C 6.1/include" -D_FORTIFY_SOURCE=2 -D__USE_MINGW_ANSI_STDIO=1 -DFD_SETSIZE=2048 -IC:/msys64/mingw32/lib/libffi-3.2.1/include -D_WIN32_WINNT=0x0501 -D__MINGW_USE_VC2005_COMPAT -D_FILE_OFFSET_BITS=64  -march=i686 -mtune=generic -O2 -pipe -IC:/msys64/mingw32/lib/libffi-3.2.1/include -DFULL_SSL_MODE_SUPPORT  conftest.c -o conftest.i"
checked program was:
/* begin */
1: #include "ruby.h"
2: 
3: #include <winsock2.h>
4: #include <windows.h>
5: #include <errmsg.h>
/* end */

--------------------

have_header: checking for mysqld_error.h... -------------------- yes

"i686-w64-mingw32-gcc -E -IE:/ruby24x86/include/ruby-2.4.0/i386-mingw32 -IE:/ruby24x86/include/ruby-2.4.0/ruby/backward -IE:/ruby24x86/include/ruby-2.4.0 -I. "-IC:/Program Files (x86)/MySQL/MySQL Connector C 6.1/include" -D_FORTIFY_SOURCE=2 -D__USE_MINGW_ANSI_STDIO=1 -DFD_SETSIZE=2048 -IC:/msys64/mingw32/lib/libffi-3.2.1/include -D_WIN32_WINNT=0x0501 -D__MINGW_USE_VC2005_COMPAT -D_FILE_OFFSET_BITS=64  -march=i686 -mtune=generic -O2 -pipe -IC:/msys64/mingw32/lib/libffi-3.2.1/include -DFULL_SSL_MODE_SUPPORT  conftest.c -o conftest.i"
checked program was:
/* begin */
1: #include "ruby.h"
2: 
3: #include <winsock2.h>
4: #include <windows.h>
5: #include <mysqld_error.h>
/* end */

--------------------

"i686-w64-mingw32-gcc -o conftest.exe -IE:/ruby24x86/include/ruby-2.4.0/i386-mingw32 -IE:/ruby24x86/include/ruby-2.4.0/ruby/backward -IE:/ruby24x86/include/ruby-2.4.0 -I. "-IC:/Program Files (x86)/MySQL/MySQL Connector C 6.1/include" -D_FORTIFY_SOURCE=2 -D__USE_MINGW_ANSI_STDIO=1 -DFD_SETSIZE=2048 -IC:/msys64/mingw32/lib/libffi-3.2.1/include -D_WIN32_WINNT=0x0501 -D__MINGW_USE_VC2005_COMPAT -D_FILE_OFFSET_BITS=64  -march=i686 -mtune=generic -O2 -pipe -IC:/msys64/mingw32/lib/libffi-3.2.1/include -DFULL_SSL_MODE_SUPPORT conftest.c  -L. -LE:/ruby24x86/lib -L"C:/Program Files (x86)/MySQL/MySQL Connector C 6.1/lib" -L. -pipe     -lmsvcrt-ruby240 -Werror -Weverything -lgmp -lshell32 -lws2_32 -liphlpapi -limagehlp -lshlwapi  "
i686-w64-mingw32-gcc: error: unrecognized command line option '-Weverything'
checked program was:
/* begin */
1: #include "ruby.h"
2: 
3: #include <winsock2.h>
4: #include <windows.h>
5: int main() {return 0;}
/* end */

"i686-w64-mingw32-gcc -o conftest.exe -IE:/ruby24x86/include/ruby-2.4.0/i386-mingw32 -IE:/ruby24x86/include/ruby-2.4.0/ruby/backward -IE:/ruby24x86/include/ruby-2.4.0 -I. "-IC:/Program Files (x86)/MySQL/MySQL Connector C 6.1/include" -D_FORTIFY_SOURCE=2 -D__USE_MINGW_ANSI_STDIO=1 -DFD_SETSIZE=2048 -IC:/msys64/mingw32/lib/libffi-3.2.1/include -D_WIN32_WINNT=0x0501 -D__MINGW_USE_VC2005_COMPAT -D_FILE_OFFSET_BITS=64  -march=i686 -mtune=generic -O2 -pipe -IC:/msys64/mingw32/lib/libffi-3.2.1/include -DFULL_SSL_MODE_SUPPORT conftest.c  -L. -LE:/ruby24x86/lib -L"C:/Program Files (x86)/MySQL/MySQL Connector C 6.1/lib" -L. -pipe     -lmsvcrt-ruby240 -Werror -Wno-bad-function-cast -lgmp -lshell32 -lws2_32 -liphlpapi -limagehlp -lshlwapi  "
checked program was:
/* begin */
1: #include "ruby.h"
2: 
3: #include <winsock2.h>
4: #include <windows.h>
5: int main() {return 0;}
/* end */

"i686-w64-mingw32-gcc -o conftest.exe -IE:/ruby24x86/include/ruby-2.4.0/i386-mingw32 -IE:/ruby24x86/include/ruby-2.4.0/ruby/backward -IE:/ruby24x86/include/ruby-2.4.0 -I. "-IC:/Program Files (x86)/MySQL/MySQL Connector C 6.1/include" -D_FORTIFY_SOURCE=2 -D__USE_MINGW_ANSI_STDIO=1 -DFD_SETSIZE=2048 -IC:/msys64/mingw32/lib/libffi-3.2.1/include -D_WIN32_WINNT=0x0501 -D__MINGW_USE_VC2005_COMPAT -D_FILE_OFFSET_BITS=64  -march=i686 -mtune=generic -O2 -pipe -IC:/msys64/mingw32/lib/libffi-3.2.1/include -DFULL_SSL_MODE_SUPPORT conftest.c  -L. -LE:/ruby24x86/lib -L"C:/Program Files (x86)/MySQL/MySQL Connector C 6.1/lib" -L. -pipe     -lmsvcrt-ruby240 -Werror -Wno-conditional-uninitialized -lgmp -lshell32 -lws2_32 -liphlpapi -limagehlp -lshlwapi  "
checked program was:
/* begin */
1: #include "ruby.h"
2: 
3: #include <winsock2.h>
4: #include <windows.h>
5: int main() {return 0;}
/* end */

"i686-w64-mingw32-gcc -o conftest.exe -IE:/ruby24x86/include/ruby-2.4.0/i386-mingw32 -IE:/ruby24x86/include/ruby-2.4.0/ruby/backward -IE:/ruby24x86/include/ruby-2.4.0 -I. "-IC:/Program Files (x86)/MySQL/MySQL Connector C 6.1/include" -D_FORTIFY_SOURCE=2 -D__USE_MINGW_ANSI_STDIO=1 -DFD_SETSIZE=2048 -IC:/msys64/mingw32/lib/libffi-3.2.1/include -D_WIN32_WINNT=0x0501 -D__MINGW_USE_VC2005_COMPAT -D_FILE_OFFSET_BITS=64  -march=i686 -mtune=generic -O2 -pipe -IC:/msys64/mingw32/lib/libffi-3.2.1/include -DFULL_SSL_MODE_SUPPORT conftest.c  -L. -LE:/ruby24x86/lib -L"C:/Program Files (x86)/MySQL/MySQL Connector C 6.1/lib" -L. -pipe     -lmsvcrt-ruby240 -Werror -Wno-covered-switch-default -lgmp -lshell32 -lws2_32 -liphlpapi -limagehlp -lshlwapi  "
checked program was:
/* begin */
1: #include "ruby.h"
2: 
3: #include <winsock2.h>
4: #include <windows.h>
5: int main() {return 0;}
/* end */

"i686-w64-mingw32-gcc -o conftest.exe -IE:/ruby24x86/include/ruby-2.4.0/i386-mingw32 -IE:/ruby24x86/include/ruby-2.4.0/ruby/backward -IE:/ruby24x86/include/ruby-2.4.0 -I. "-IC:/Program Files (x86)/MySQL/MySQL Connector C 6.1/include" -D_FORTIFY_SOURCE=2 -D__USE_MINGW_ANSI_STDIO=1 -DFD_SETSIZE=2048 -IC:/msys64/mingw32/lib/libffi-3.2.1/include -D_WIN32_WINNT=0x0501 -D__MINGW_USE_VC2005_COMPAT -D_FILE_OFFSET_BITS=64  -march=i686 -mtune=generic -O2 -pipe -IC:/msys64/mingw32/lib/libffi-3.2.1/include -DFULL_SSL_MODE_SUPPORT conftest.c  -L. -LE:/ruby24x86/lib -L"C:/Program Files (x86)/MySQL/MySQL Connector C 6.1/lib" -L. -pipe     -lmsvcrt-ruby240 -Werror -Wno-declaration-after-statement -lgmp -lshell32 -lws2_32 -liphlpapi -limagehlp -lshlwapi  "
checked program was:
/* begin */
1: #include "ruby.h"
2: 
3: #include <winsock2.h>
4: #include <windows.h>
5: int main() {return 0;}
/* end */

"i686-w64-mingw32-gcc -o conftest.exe -IE:/ruby24x86/include/ruby-2.4.0/i386-mingw32 -IE:/ruby24x86/include/ruby-2.4.0/ruby/backward -IE:/ruby24x86/include/ruby-2.4.0 -I. "-IC:/Program Files (x86)/MySQL/MySQL Connector C 6.1/include" -D_FORTIFY_SOURCE=2 -D__USE_MINGW_ANSI_STDIO=1 -DFD_SETSIZE=2048 -IC:/msys64/mingw32/lib/libffi-3.2.1/include -D_WIN32_WINNT=0x0501 -D__MINGW_USE_VC2005_COMPAT -D_FILE_OFFSET_BITS=64  -march=i686 -mtune=generic -O2 -pipe -IC:/msys64/mingw32/lib/libffi-3.2.1/include -DFULL_SSL_MODE_SUPPORT conftest.c  -L. -LE:/ruby24x86/lib -L"C:/Program Files (x86)/MySQL/MySQL Connector C 6.1/lib" -L. -pipe     -lmsvcrt-ruby240 -Werror -Wno-disabled-macro-expansion -lgmp -lshell32 -lws2_32 -liphlpapi -limagehlp -lshlwapi  "
checked program was:
/* begin */
1: #include "ruby.h"
2: 
3: #include <winsock2.h>
4: #include <windows.h>
5: int main() {return 0;}
/* end */

"i686-w64-mingw32-gcc -o conftest.exe -IE:/ruby24x86/include/ruby-2.4.0/i386-mingw32 -IE:/ruby24x86/include/ruby-2.4.0/ruby/backward -IE:/ruby24x86/include/ruby-2.4.0 -I. "-IC:/Program Files (x86)/MySQL/MySQL Connector C 6.1/include" -D_FORTIFY_SOURCE=2 -D__USE_MINGW_ANSI_STDIO=1 -DFD_SETSIZE=2048 -IC:/msys64/mingw32/lib/libffi-3.2.1/include -D_WIN32_WINNT=0x0501 -D__MINGW_USE_VC2005_COMPAT -D_FILE_OFFSET_BITS=64  -march=i686 -mtune=generic -O2 -pipe -IC:/msys64/mingw32/lib/libffi-3.2.1/include -DFULL_SSL_MODE_SUPPORT conftest.c  -L. -LE:/ruby24x86/lib -L"C:/Program Files (x86)/MySQL/MySQL Connector C 6.1/lib" -L. -pipe     -lmsvcrt-ruby240 -Werror -Wno-documentation-unknown-command -lgmp -lshell32 -lws2_32 -liphlpapi -limagehlp -lshlwapi  "
checked program was:
/* begin */
1: #include "ruby.h"
2: 
3: #include <winsock2.h>
4: #include <windows.h>
5: int main() {return 0;}
/* end */

"i686-w64-mingw32-gcc -o conftest.exe -IE:/ruby24x86/include/ruby-2.4.0/i386-mingw32 -IE:/ruby24x86/include/ruby-2.4.0/ruby/backward -IE:/ruby24x86/include/ruby-2.4.0 -I. "-IC:/Program Files (x86)/MySQL/MySQL Connector C 6.1/include" -D_FORTIFY_SOURCE=2 -D__USE_MINGW_ANSI_STDIO=1 -DFD_SETSIZE=2048 -IC:/msys64/mingw32/lib/libffi-3.2.1/include -D_WIN32_WINNT=0x0501 -D__MINGW_USE_VC2005_COMPAT -D_FILE_OFFSET_BITS=64  -march=i686 -mtune=generic -O2 -pipe -IC:/msys64/mingw32/lib/libffi-3.2.1/include -DFULL_SSL_MODE_SUPPORT conftest.c  -L. -LE:/ruby24x86/lib -L"C:/Program Files (x86)/MySQL/MySQL Connector C 6.1/lib" -L. -pipe     -lmsvcrt-ruby240 -Werror -Wno-missing-field-initializers -lgmp -lshell32 -lws2_32 -liphlpapi -limagehlp -lshlwapi  "
checked program was:
/* begin */
1: #include "ruby.h"
2: 
3: #include <winsock2.h>
4: #include <windows.h>
5: int main() {return 0;}
/* end */

"i686-w64-mingw32-gcc -o conftest.exe -IE:/ruby24x86/include/ruby-2.4.0/i386-mingw32 -IE:/ruby24x86/include/ruby-2.4.0/ruby/backward -IE:/ruby24x86/include/ruby-2.4.0 -I. "-IC:/Program Files (x86)/MySQL/MySQL Connector C 6.1/include" -D_FORTIFY_SOURCE=2 -D__USE_MINGW_ANSI_STDIO=1 -DFD_SETSIZE=2048 -IC:/msys64/mingw32/lib/libffi-3.2.1/include -D_WIN32_WINNT=0x0501 -D__MINGW_USE_VC2005_COMPAT -D_FILE_OFFSET_BITS=64  -march=i686 -mtune=generic -O2 -pipe -IC:/msys64/mingw32/lib/libffi-3.2.1/include -DFULL_SSL_MODE_SUPPORT conftest.c  -L. -LE:/ruby24x86/lib -L"C:/Program Files (x86)/MySQL/MySQL Connector C 6.1/lib" -L. -pipe     -lmsvcrt-ruby240 -Werror -Wno-missing-variable-declarations -lgmp -lshell32 -lws2_32 -liphlpapi -limagehlp -lshlwapi  "
checked program was:
/* begin */
1: #include "ruby.h"
2: 
3: #include <winsock2.h>
4: #include <windows.h>
5: int main() {return 0;}
/* end */

"i686-w64-mingw32-gcc -o conftest.exe -IE:/ruby24x86/include/ruby-2.4.0/i386-mingw32 -IE:/ruby24x86/include/ruby-2.4.0/ruby/backward -IE:/ruby24x86/include/ruby-2.4.0 -I. "-IC:/Program Files (x86)/MySQL/MySQL Connector C 6.1/include" -D_FORTIFY_SOURCE=2 -D__USE_MINGW_ANSI_STDIO=1 -DFD_SETSIZE=2048 -IC:/msys64/mingw32/lib/libffi-3.2.1/include -D_WIN32_WINNT=0x0501 -D__MINGW_USE_VC2005_COMPAT -D_FILE_OFFSET_BITS=64  -march=i686 -mtune=generic -O2 -pipe -IC:/msys64/mingw32/lib/libffi-3.2.1/include -DFULL_SSL_MODE_SUPPORT conftest.c  -L. -LE:/ruby24x86/lib -L"C:/Program Files (x86)/MySQL/MySQL Connector C 6.1/lib" -L. -pipe     -lmsvcrt-ruby240 -Werror -Wno-padded -lgmp -lshell32 -lws2_32 -liphlpapi -limagehlp -lshlwapi  "
checked program was:
/* begin */
1: #include "ruby.h"
2: 
3: #include <winsock2.h>
4: #include <windows.h>
5: int main() {return 0;}
/* end */

"i686-w64-mingw32-gcc -o conftest.exe -IE:/ruby24x86/include/ruby-2.4.0/i386-mingw32 -IE:/ruby24x86/include/ruby-2.4.0/ruby/backward -IE:/ruby24x86/include/ruby-2.4.0 -I. "-IC:/Program Files (x86)/MySQL/MySQL Connector C 6.1/include" -D_FORTIFY_SOURCE=2 -D__USE_MINGW_ANSI_STDIO=1 -DFD_SETSIZE=2048 -IC:/msys64/mingw32/lib/libffi-3.2.1/include -D_WIN32_WINNT=0x0501 -D__MINGW_USE_VC2005_COMPAT -D_FILE_OFFSET_BITS=64  -march=i686 -mtune=generic -O2 -pipe -IC:/msys64/mingw32/lib/libffi-3.2.1/include -DFULL_SSL_MODE_SUPPORT conftest.c  -L. -LE:/ruby24x86/lib -L"C:/Program Files (x86)/MySQL/MySQL Connector C 6.1/lib" -L. -pipe     -lmsvcrt-ruby240 -Werror -Wno-reserved-id-macro -lgmp -lshell32 -lws2_32 -liphlpapi -limagehlp -lshlwapi  "
checked program was:
/* begin */
1: #include "ruby.h"
2: 
3: #include <winsock2.h>
4: #include <windows.h>
5: int main() {return 0;}
/* end */

"i686-w64-mingw32-gcc -o conftest.exe -IE:/ruby24x86/include/ruby-2.4.0/i386-mingw32 -IE:/ruby24x86/include/ruby-2.4.0/ruby/backward -IE:/ruby24x86/include/ruby-2.4.0 -I. "-IC:/Program Files (x86)/MySQL/MySQL Connector C 6.1/include" -D_FORTIFY_SOURCE=2 -D__USE_MINGW_ANSI_STDIO=1 -DFD_SETSIZE=2048 -IC:/msys64/mingw32/lib/libffi-3.2.1/include -D_WIN32_WINNT=0x0501 -D__MINGW_USE_VC2005_COMPAT -D_FILE_OFFSET_BITS=64  -march=i686 -mtune=generic -O2 -pipe -IC:/msys64/mingw32/lib/libffi-3.2.1/include -DFULL_SSL_MODE_SUPPORT conftest.c  -L. -LE:/ruby24x86/lib -L"C:/Program Files (x86)/MySQL/MySQL Connector C 6.1/lib" -L. -pipe     -lmsvcrt-ruby240 -Werror -Wno-sign-conversion -lgmp -lshell32 -lws2_32 -liphlpapi -limagehlp -lshlwapi  "
checked program was:
/* begin */
1: #include "ruby.h"
2: 
3: #include <winsock2.h>
4: #include <windows.h>
5: int main() {return 0;}
/* end */

"i686-w64-mingw32-gcc -o conftest.exe -IE:/ruby24x86/include/ruby-2.4.0/i386-mingw32 -IE:/ruby24x86/include/ruby-2.4.0/ruby/backward -IE:/ruby24x86/include/ruby-2.4.0 -I. "-IC:/Program Files (x86)/MySQL/MySQL Connector C 6.1/include" -D_FORTIFY_SOURCE=2 -D__USE_MINGW_ANSI_STDIO=1 -DFD_SETSIZE=2048 -IC:/msys64/mingw32/lib/libffi-3.2.1/include -D_WIN32_WINNT=0x0501 -D__MINGW_USE_VC2005_COMPAT -D_FILE_OFFSET_BITS=64  -march=i686 -mtune=generic -O2 -pipe -IC:/msys64/mingw32/lib/libffi-3.2.1/include -DFULL_SSL_MODE_SUPPORT conftest.c  -L. -LE:/ruby24x86/lib -L"C:/Program Files (x86)/MySQL/MySQL Connector C 6.1/lib" -L. -pipe     -lmsvcrt-ruby240 -Werror -Wno-static-in-inline -lgmp -lshell32 -lws2_32 -liphlpapi -limagehlp -lshlwapi  "
checked program was:
/* begin */
1: #include "ruby.h"
2: 
3: #include <winsock2.h>
4: #include <windows.h>
5: int main() {return 0;}
/* end */

"i686-w64-mingw32-gcc -o conftest.exe -IE:/ruby24x86/include/ruby-2.4.0/i386-mingw32 -IE:/ruby24x86/include/ruby-2.4.0/ruby/backward -IE:/ruby24x86/include/ruby-2.4.0 -I. "-IC:/Program Files (x86)/MySQL/MySQL Connector C 6.1/include" -D_FORTIFY_SOURCE=2 -D__USE_MINGW_ANSI_STDIO=1 -DFD_SETSIZE=2048 -IC:/msys64/mingw32/lib/libffi-3.2.1/include -D_WIN32_WINNT=0x0501 -D__MINGW_USE_VC2005_COMPAT -D_FILE_OFFSET_BITS=64  -march=i686 -mtune=generic -O2 -pipe -IC:/msys64/mingw32/lib/libffi-3.2.1/include -DFULL_SSL_MODE_SUPPORT conftest.c  -L. -LE:/ruby24x86/lib -L"C:/Program Files (x86)/MySQL/MySQL Connector C 6.1/lib" -L. -pipe     -lmsvcrt-ruby240 -Werror -Wno-switch-enum -lgmp -lshell32 -lws2_32 -liphlpapi -limagehlp -lshlwapi  "
checked program was:
/* begin */
1: #include "ruby.h"
2: 
3: #include <winsock2.h>
4: #include <windows.h>
5: int main() {return 0;}
/* end */

"i686-w64-mingw32-gcc -o conftest.exe -IE:/ruby24x86/include/ruby-2.4.0/i386-mingw32 -IE:/ruby24x86/include/ruby-2.4.0/ruby/backward -IE:/ruby24x86/include/ruby-2.4.0 -I. "-IC:/Program Files (x86)/MySQL/MySQL Connector C 6.1/include" -D_FORTIFY_SOURCE=2 -D__USE_MINGW_ANSI_STDIO=1 -DFD_SETSIZE=2048 -IC:/msys64/mingw32/lib/libffi-3.2.1/include -D_WIN32_WINNT=0x0501 -D__MINGW_USE_VC2005_COMPAT -D_FILE_OFFSET_BITS=64  -march=i686 -mtune=generic -O2 -pipe -IC:/msys64/mingw32/lib/libffi-3.2.1/include -DFULL_SSL_MODE_SUPPORT conftest.c  -L. -LE:/ruby24x86/lib -L"C:/Program Files (x86)/MySQL/MySQL Connector C 6.1/lib" -L. -pipe     -lmsvcrt-ruby240 -Werror -Wno-undef -lgmp -lshell32 -lws2_32 -liphlpapi -limagehlp -lshlwapi  "
checked program was:
/* begin */
1: #include "ruby.h"
2: 
3: #include <winsock2.h>
4: #include <windows.h>
5: int main() {return 0;}
/* end */

"i686-w64-mingw32-gcc -o conftest.exe -IE:/ruby24x86/include/ruby-2.4.0/i386-mingw32 -IE:/ruby24x86/include/ruby-2.4.0/ruby/backward -IE:/ruby24x86/include/ruby-2.4.0 -I. "-IC:/Program Files (x86)/MySQL/MySQL Connector C 6.1/include" -D_FORTIFY_SOURCE=2 -D__USE_MINGW_ANSI_STDIO=1 -DFD_SETSIZE=2048 -IC:/msys64/mingw32/lib/libffi-3.2.1/include -D_WIN32_WINNT=0x0501 -D__MINGW_USE_VC2005_COMPAT -D_FILE_OFFSET_BITS=64  -march=i686 -mtune=generic -O2 -pipe -IC:/msys64/mingw32/lib/libffi-3.2.1/include -DFULL_SSL_MODE_SUPPORT conftest.c  -L. -LE:/ruby24x86/lib -L"C:/Program Files (x86)/MySQL/MySQL Connector C 6.1/lib" -L. -pipe     -lmsvcrt-ruby240 -Werror -Wno-unreachable-code -lgmp -lshell32 -lws2_32 -liphlpapi -limagehlp -lshlwapi  "
checked program was:
/* begin */
1: #include "ruby.h"
2: 
3: #include <winsock2.h>
4: #include <windows.h>
5: int main() {return 0;}
/* end */

"i686-w64-mingw32-gcc -o conftest.exe -IE:/ruby24x86/include/ruby-2.4.0/i386-mingw32 -IE:/ruby24x86/include/ruby-2.4.0/ruby/backward -IE:/ruby24x86/include/ruby-2.4.0 -I. "-IC:/Program Files (x86)/MySQL/MySQL Connector C 6.1/include" -D_FORTIFY_SOURCE=2 -D__USE_MINGW_ANSI_STDIO=1 -DFD_SETSIZE=2048 -IC:/msys64/mingw32/lib/libffi-3.2.1/include -D_WIN32_WINNT=0x0501 -D__MINGW_USE_VC2005_COMPAT -D_FILE_OFFSET_BITS=64  -march=i686 -mtune=generic -O2 -pipe -IC:/msys64/mingw32/lib/libffi-3.2.1/include -DFULL_SSL_MODE_SUPPORT conftest.c  -L. -LE:/ruby24x86/lib -L"C:/Program Files (x86)/MySQL/MySQL Connector C 6.1/lib" -L. -pipe     -lmsvcrt-ruby240 -Werror -Wno-used-but-marked-unused -lgmp -lshell32 -lws2_32 -liphlpapi -limagehlp -lshlwapi  "
checked program was:
/* begin */
1: #include "ruby.h"
2: 
3: #include <winsock2.h>
4: #include <windows.h>
5: int main() {return 0;}
/* end */
```

--- a/Gemfile
+++ b/Gemfile
@@ -3,7 +3,7 @@ source 'https://rubygems.org'
 gemspec
 
 gem 'rake', '~> 10.4.2'
-gem 'rake-compiler', '~> 0.9.5'
+gem 'rake-compiler', '~> 1.0'
 
 group :test do
   gem 'eventmachine' unless RUBY_PLATFORM =~ /mswin|mingw/
@@ -22,7 +22,7 @@ end
 
 group :development do
   gem 'pry'
-  gem 'rake-compiler-dock', '~> 0.5.1'
+  gem 'rake-compiler-dock', '~> 0.6.0'
 end
 
 platforms :rbx do

--- a/appveyor.yml
+++ b/appveyor.yml
@@ -2,12 +2,14 @@
 version: "{build}"
 clone_depth: 10
 install:
+  - SET PATH=C:\MinGW\msys\1.0\bin;%PATH%
   - SET PATH=C:\Ruby%ruby_version%\bin;%PATH%
   - ruby --version
   - gem --version
   - gem install bundler --quiet --no-ri --no-rdoc
   - bundler --version
   - bundle install --without benchmarks --path vendor/bundle
+  - IF DEFINED MINGW_PACKAGE_PREFIX (ridk exec pacman -S --noconfirm --needed %MINGW_PACKAGE_PREFIX%-libmariadbclient)
 build_script:
   - bundle exec rake compile
 test_script:
@@ -20,11 +22,14 @@ test_script:
     FLUSH PRIVILEGES;
     "
   - bundle exec rake spec
-# Where do I get Unix find?
-#on_failure:
-#  - find tmp -name "*.log" -exec cat {};
+on_failure:
+  - find tmp -name "*.log" | xargs cat
 environment:
   matrix:
+    - ruby_version: "24-x64"
+      MINGW_PACKAGE_PREFIX: "mingw-w64-x86_64"
+    - ruby_version: "24"
+      MINGW_PACKAGE_PREFIX: "mingw-w64-i686"
     - ruby_version: "23-x64"
     - ruby_version: "23"
     - ruby_version: "22-x64"

--- a/ext/mysql2/extconf.rb
+++ b/ext/mysql2/extconf.rb
@@ -92,7 +92,7 @@ elsif (mc = (with_config('mysql-config') || Dir[GLOB].first))
 else
   _, usr_local_lib = dir_config('mysql', '/usr/local')
 
-  asplode("mysql client") unless find_library('mysqlclient', 'mysql_query', usr_local_lib, "#{usr_local_lib}/mysql")
+  asplode("mysql client") unless find_library('mysqlclient', nil, usr_local_lib, "#{usr_local_lib}/mysql")
 
   rpath_dir = usr_local_lib
 end
@@ -177,7 +177,7 @@ unless enabled_sanitizers.empty?
   $CFLAGS << ' -g -fno-omit-frame-pointer'
 end
 
-if RUBY_PLATFORM =~ /mswin|mingw/
+if RUBY_PLATFORM =~ /mswin|mingw/ && !defined?(RubyInstaller)
   # Build libmysql.a interface link library
   require 'rake'
 

--- a/lib/mysql2.rb
+++ b/lib/mysql2.rb
@@ -13,16 +13,20 @@ if RUBY_PLATFORM =~ /mswin|mingw/
     ENV['RUBY_MYSQL2_LIBMYSQL_DLL']
   elsif File.exist?(File.expand_path('../vendor/libmysql.dll', File.dirname(__FILE__)))
     # Use vendor/libmysql.dll if it exists, convert slashes for Win32 LoadLibrary
-    File.expand_path('../vendor/libmysql.dll', File.dirname(__FILE__)).tr('/', '\\')
+    File.expand_path('../vendor/libmysql.dll', File.dirname(__FILE__))
+  elsif defined?(RubyInstaller)
+    # RubyInstaller-2.4+ native build doesn't need DLL preloading
   else
     # This will use default / system library paths
     'libmysql.dll'
   end
 
-  require 'Win32API'
-  LoadLibrary = Win32API.new('Kernel32', 'LoadLibrary', ['P'], 'I')
-  if 0 == LoadLibrary.call(dll_path)
-    abort "Failed to load libmysql.dll from #{dll_path}"
+  if dll_path
+    require 'Win32API'
+    LoadLibrary = Win32API.new('Kernel32', 'LoadLibrary', ['P'], 'I')
+    if 0 == LoadLibrary.call(dll_path)
+      abort "Failed to load libmysql.dll from #{dll_path}"
+    end
   end
 end
 

--- a/mysql2.gemspec
+++ b/mysql2.gemspec
@@ -13,4 +13,6 @@ Mysql2::GEMSPEC = Gem::Specification.new do |s|
 
   s.files = `git ls-files README.md CHANGELOG.md LICENSE ext lib support`.split
   s.test_files = `git ls-files spec examples`.split
+
+  s.metadata['msys2_mingw_dependencies'] = 'libmariadbclient'
 end

--- a/spec/mysql2/statement_spec.rb
+++ b/spec/mysql2/statement_spec.rb
@@ -712,7 +712,6 @@ RSpec.describe Mysql2::Statement do
 
     it 'should return number of rows affected by an insert' do
       stmt = @client.prepare 'INSERT INTO lastIdTest (blah) VALUES (?)'
-      expect(stmt.affected_rows).to eq 0
       stmt.execute 1
       expect(stmt.affected_rows).to eq 1
     end

--- a/tasks/compile.rake
+++ b/tasks/compile.rake
@@ -9,7 +9,7 @@ Rake::ExtensionTask.new("mysql2", Mysql2::GEMSPEC) do |ext|
   # clean compiled extension
   CLEAN.include "#{ext.lib_dir}/*.#{RbConfig::CONFIG['DLEXT']}"
 
-  if RUBY_PLATFORM =~ /mswin|mingw/
+  if RUBY_PLATFORM =~ /mswin|mingw/ && !defined?(RubyInstaller)
     # Expand the path because the build dir is 3-4 levels deep in tmp/platform/version/
     connector_dir = File.expand_path("../../vendor/#{vendor_mysql_dir}", __FILE__)
     ext.config_options = ["--with-mysql-dir=#{connector_dir}"]
@@ -26,6 +26,10 @@ Rake::ExtensionTask.new("mysql2", Mysql2::GEMSPEC) do |ext|
       Rake::Task['lib/mysql2/mysql2.rb'].invoke
       # vendor/libmysql.dll is invoked from extconf.rb
       Rake::Task['vendor/README'].invoke
+
+      # only the source gem has a package dependency - the binary gem ships it's own DLL version
+      spec.metadata.delete('msys2_mingw_dependencies')
+
       spec.files << 'lib/mysql2/mysql2.rb'
       spec.files << 'vendor/libmysql.dll'
       spec.files << 'vendor/README'
@@ -77,7 +81,7 @@ task :devkit do
 end
 
 if RUBY_PLATFORM =~ /mingw|mswin/
-  Rake::Task['compile'].prerequisites.unshift 'vendor:mysql'
+  Rake::Task['compile'].prerequisites.unshift 'vendor:mysql' unless defined?(RubyInstaller)
   Rake::Task['compile'].prerequisites.unshift 'devkit'
 else
   if Rake::Task.tasks.map(&:name).include? 'cross'

